torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp

```diff
@@ -41,6 +89,52 @@ Expr* Val::getOrigin() {
   return (fusion_->origin(this));
 }
 
+void Scope::insert_before(Expr* ref, Expr* expr) {
```

Comment:
Is there an iterator increment missing? Also, if you don't match, won't it insert before the first element?

Reply:
This comment applies to insert_after and erase as well.